### PR TITLE
CEDS-1344 Add Seal and CarrierDetails missing data

### PR DESCRIPTION
**Pull request by PauloPortugal**
* Add seal to the new cache model
* Add CarrierDetails to new cache model
* Format code using Scalafmt

### CODE REVIEW

File: app/controllers/declaration/CarrierDetailsPageController.scala [hunk 1]
```diff
@@ -55,9 +57,16 @@ class CarrierDetailsPageController @Inject()(
       .fold(
         (formWithErrors: Form[CarrierDetails]) => Future.successful(BadRequest(carrierDetailsPage(formWithErrors))),
         form =>
-          customsCacheService.cache[CarrierDetails](cacheId, CarrierDetails.id, form).map { _ =>
-            Redirect(controllers.declaration.routes.DeclarationAdditionalActorsController.displayForm())
-        }
+          for {
```
Comment (ed0906) [inline app/controllers/declaration/CarrierDetailsPageController.scala]: If you pull the for comprehension into the `updateCache` method (see LocationController) it means less refactoring when we switch the retrieves over

File: app/controllers/declaration/SealController.scala [hunk 2]
```diff
@@ -108,12 +108,16 @@ class SealController @Inject()(
...
       .map(data => BadRequest(sealPage(formWithErrors, cachedSeals, data.fold(false)(_.container))))
 
   private def cacheAndRedirect(seals: Seq[Seal])(implicit request: JourneyRequest[_]): Future[Result] =
-    cacheService
-      .cache[Seq[Seal]](cacheId, formId, seals)
-      .map(_ => Redirect(routes.SealController.displayForm()))
+    for {
+      _ <- updateCache(journeySessionId, seals)
+      _ <- customsCacheService.cache[Seq[Seal]](cacheId, formId, seals)
+    } yield Redirect(routes.SealController.displayForm())
```
Comment (ed0906) [inline app/controllers/declaration/SealController.scala]: similar to the above, you can move the for comprehension into `updateCache` (removes the duplication and makes  it easier to remove the old cache

File: test/controllers/declaration/CarrierDetailsPageControllerSpec.scala [hunk 3]
```diff
@@ -29,17 +31,22 @@ class CarrierDetailsPageControllerSpec extends CustomExportsBaseSpec {
...
+    Mockito.reset(mockExportsCacheService)
+
   "Carrier Details Page Controller on GET" should {
 
     "return 200 status code" in {
       val result = route(app, getRequest(uri)).get
 
       status(result) must be(OK)
-    }
+    }GoodsShipmentBuilderSpec.scala
```
Comment (ed0906) [inline test/controllers/declaration/CarrierDetailsPageControllerSpec.scala]: What's this?

File: test/controllers/declaration/SealControllerSpec.scala [hunk 4]
```diff
@@ -113,61 +115,40 @@ class SealControllerSpec extends CustomExportsBaseSpec with Generators with Prop
...
-              any(),
-              any()
-            )
-        }
+        val body = Seq(("id", "12345")) :+ addActionUrlEncoded
+        val result = route(app, postRequestFormUrlEncoded(uri, body: _*)).value
+        status(result) must be(SEE_OTHER)
+        verify(mockCustomsCacheService)
+          .cache[Seq[Seal]](any(), ArgumentMatchers.eq(Seal.formId), any())(any(), any(), any())
+        theCacheModelUpdated.seals.size mustBe 1
```
Comment (ed0906) [inline test/controllers/declaration/SealControllerSpec.scala]: Maybe a verify what the cache seal contents is?

`theCacheModelUpdated.seals(0) mustBe ...`

File: test/controllers/declaration/CarrierDetailsPageControllerSpec.scala (hunk 3 above)
Comment (PauloPortugal) [reply test/controllers/declaration/CarrierDetailsPageControllerSpec.scala]: I fat fingered that.... My bad

Comment (PauloPortugal) [reply test/controllers/declaration/CarrierDetailsPageControllerSpec.scala]: this has been addressed

File: app/controllers/declaration/SealController.scala (hunk 2 above)
Comment (PauloPortugal) [reply app/controllers/declaration/SealController.scala]: This has been addressed!

File: app/controllers/declaration/CarrierDetailsPageController.scala (hunk 1 above)
Comment (PauloPortugal) [reply app/controllers/declaration/CarrierDetailsPageController.scala]: This has been addressed!

File: test/controllers/declaration/SealControllerSpec.scala (hunk 4 above)
Comment (PauloPortugal) [reply test/controllers/declaration/SealControllerSpec.scala]: This has been addressed

**Pull request merged**